code/modules/projectiles/guns/energy/modular.dm

```diff
@@ -30,21 +30,48 @@
 	var/named = 0
 	var/described = 0
 
+/obj/item/gun/energy/laser/prototype/Initialize(mapload, var/set_icon_state, var/size, var/set_capacitor, var/set_focusing_lens, var/set_modulator, var/list/set_gun_mods)
+	. = ..()
+	icon_state = set_icon_state
+	modifystate = set_icon_state
+	origin_chassis = size
+	capacitor = set_capacitor
+	capacitor.forceMove(src)
```

Comment:
do you actually need to force_move all this stuff? can't you just not give it a location